staging/src/k8s.io/cloud-provider/controllers/node/node_controller_test.go

```diff
@@ -1294,10 +1294,18 @@ func Test_AddCloudNode(t *testing.T) {
 				recorder:                  eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "cloud-node-controller"}),
 				nodeStatusUpdateFrequency: 1 * time.Second,
 			}
+
+			// activate node informer
+			factory.Core().V1().Nodes().Informer()
+			factory.Start(nil)
+			factory.WaitForCacheSync(nil)
+
+			cloudNodeController.nodesLister = factory.Core().V1().Nodes().Lister()
+
```

Comment:
@andrewsykim I use this way to set up the lister, not sure if this is correct but I got unit tests passing, let me know if I need to improve anything :)

Reply:
Looks correct but I think it can be a bit cleaner by setting node lister when `CloudNodeController` is initialized. I believe calling `factory.Core().V1().Nodes().Lister()` will also activate the informer.
```go
			cloudNodeController := &CloudNodeController{
				kubeClient:                clientset,
				nodeInformer:              factory.Core().V1().Nodes(),
                                nodeLister:               factory.Core().V1().Nodes().Lister(),
				 cloud:                     test.fakeCloud,
				recorder:                  eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "cloud-node-controller"}),
				nodeStatusUpdateFrequency: 1 * time.Second,
			}

			factory.Start(nil)
			factory.WaitForCacheSync(nil)
```